api/sessionrecording/session_recording.go

```diff
@@ -0,0 +1,303 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sessionrecording
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"io"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/defaults"
+	apievents "github.com/gravitational/teleport/api/types/events"
+)
+
+const (
+	// Int32Size is a constant for 32 bit integer byte size
+	Int32Size = 4
+
+	// Int64Size is a constant for 64 bit integer byte size
+	Int64Size = 8
+
+	// MaxProtoMessageSizeBytes is maximum protobuf marshaled message size
+	MaxProtoMessageSizeBytes = 64 * 1024
+
+	// ProtoStreamV1 is a version of the binary protocol
+	ProtoStreamV1 = 1
+
+	// ProtoStreamV1PartHeaderSize is the size of the part of the protocol stream
+	// on disk format, it consists of
+	// * 8 bytes for the format version
+	// * 8 bytes for meaningful size of the part
+	// * 8 bytes for optional padding size at the end of the slice
+	ProtoStreamV1PartHeaderSize = Int64Size * 3
+
+	// ProtoStreamV1RecordHeaderSize is the size of the header
+	// of the record header, it consists of the record length
+	ProtoStreamV1RecordHeaderSize = Int32Size
+)
+
+// NewReader returns a new reader with slice pool
```

Comment:
```suggestion
// NewReader returns a new session recording reader.
```
I wouldn't mention slice pool.

Reply:
good call, removed